digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/lane/LaneService.scala

```diff
@@ -518,6 +517,16 @@ trait LaneOperations {
 
   }
 
+  def getRoadAddressForLanes(lanes: Seq[PieceWiseLane]): Seq[PieceWiseLane] = {
+    val (lanesWithUpdatedInfo, lanesMissingUpdatedInfo) = LogUtils.time(logger, "TEST LOG Get Viite road address for lanes") {
+      roadAddressService.laneWithRoadAddress(lanes).partition(_.attributes.contains("VIITE_ROAD_NUMBER"))
```

Comment:
laneWithRoadAddress on kyl outo nimi, Miksi roadAddressService pitäisi tietää mitään kaistoista? Se hakee tieosoitteet ja laneService sitten mappaa sen oikeaan kaistaan.

Reply:
Niin sinäänsä voisi erottaa näin vastuut selkeämmiksi. Ei nyt ehkä kovin väärin nykyinenkään toteutus mielestäni

Reply:
hyym omasta mielestä vastuut menee väärin. Ei nyt maailmaa kaada mutta itse en tykkää.

Reply:
Liittyen vähän design periaateeseen että service käsittelee vaan yhtä asiaa. Jos pitää tehdä enemmän kuin yhtä asiaa pitäisi olla erillinen taso sille jossa kaksi erillistä asiaa tuodaan yhteen. Voisi olla joku erillinen business logic taso johon injectoidaan ne service joita tarvitaan kuhunkin prosessiin. Tai jokin event buss systeemi joka kokoaa palveluita yhdeksi kokonaisuuseksi.

Reply:
Kai vois olla joku business prosess engine systeemi.